agent/command/s3_put_test.go

```diff
@@ -494,24 +493,51 @@ func TestPreservePath(t *testing.T) {
 	require.NoError(t, s.Execute(ctx, comm, logger, conf))
 	it, err := s.bucket.List(ctx, "")
 	require.NoError(t, err)
-	expected := map[string]bool{
+
+	expected := getExpected(false)
+	expectedWindows := getExpected(true)
+
+	for it.Next(ctx) {
+		expected[it.Item().Name()] = true
+		expectedWindows[it.Item().Name()] = true
+	}
+
+	trueForExpected := true
+	trueForExpectedWindows := true
+
+	for _, exists := range expected {
+		if !exists {
+			trueForExpected = false
+		}
+	}
+	for _, exists := range expectedWindows {
+		if !exists {
+			trueForExpectedWindows = false
+		}
+	}
+
+	require.True(t, trueForExpected || trueForExpectedWindows)
+
+}
+
+func getExpected(isWindows bool) map[string]bool {
+	if isWindows {
+		return map[string]bool{
+			"remote\\foo":                               false,
```

Comment:
Rather than having two separate maps with basically the same files, this could just be `filepath.Join("remote", "foo"): false` (and so on) so we don't have to care about if it's Windows or not.